internal/validators/namespace.go

```diff
@@ -107,6 +118,46 @@ func (v *Namespace) handle(req *nsRequest) admission.Response {
 	return allow("")
 }
 
+// illegalTreeLabel checks if tree labels are being created or modified
+// by any user or service account since only HNC service account is
+// allowed to do so
+func (v *Namespace) illegalTreeLabel(req *nsRequest) admission.Response {
+	msg := "Only HNC Service account can modify tree labels"
+
+	if req.op == k8sadm.Delete {
+		return allow("")
+	}
+
```

Comment:
Hmm, I had a closer look at the caller and it looks like we only call this function for Create or Update. So you can delete this.